spectralcluster/utils.py

```diff
@@ -39,7 +40,7 @@ def compute_sorted_eigenvectors(A):
            eigenvalue
     """
     # Eigen decomposition.
-    eigenvalues, eigenvectors = np.linalg.eig(A)
+    eigenvalues, eigenvectors = scipy.linalg.eigh(A, turbo=True)
```

Comment:
I think this should be `eig` instead of `eigh`.

Depending on the choice and order of the refinement operations, the enhanced affinity matrix does not have to be symmetric.

See the documentation: `Note that if input matrices are not symmetric or hermitian, no error is reported but results will be wrong.`

https://docs.scipy.org/doc/scipy/reference/generated/scipy.linalg.eigh.html#scipy.linalg.eigh

Reply:
Oh, I thought the affinity matrix is always symmetric. I was wrong.